File: public/upscale.py
import cv2
from cv2 import dnn_superres
import os

def upscale_image(image_path, model_path, output_path):
    print(f"Upscaling {image_path} to {output_path}...")
    sr = dnn_superres.DnnSuperResImpl_create()
    img = cv2.imread(image_path)
    if img is None:
        print(f"Failed to load {image_path}")
        return
    
    sr.readModel(model_path)
    sr.setModel("fsrcnn", 4)
    
    result = sr.upsample(img)
    cv2.imwrite(output_path, result, [cv2.IMWRITE_JPEG_QUALITY, 95])
    print(f"Saved {output_path}")

images = [
    'lc-coupe.jpg',
    'ls-sedan.jpg',
    'lx-mountains.jpg',
    'red-lexus.jpg',
    'isf.jpg',
    'gsf.jpg',
    'gx-series.jpg',
    'rx-series.jpg',
    'nx-series.jpg',
    'bridge-lexus.jpg'
]

model_path = "FSRCNN_x4.pb"
for img_name in images:
    if os.path.exists(img_name):
        base, ext = os.path.splitext(img_name)
        output_name = f"{base}-hq{ext}"
        upscale_image(img_name, model_path, output_name)
    else:
        print(f"{img_name} not found")

File: public/upscale.js
const sharp = require('sharp');
const fs = require('fs');

const images = [
    'lc-coupe.jpg',
    'ls-sedan.jpg',
    'lx-mountains.jpg',
    'red-lexus.jpg',
    'isf.jpg',
    'gsf.jpg',
    'gx-series.jpg',
    'rx-series.jpg',
    'nx-series.jpg',
    'bridge-lexus.jpg'
];

async function upscale() {
    for (const img of images) {
        if (fs.existsSync(img)) {
            const outPath = img.replace('.jpg', '-hq.jpg');
            console.log(`Upscaling ${img} to ${outPath}...`);
            const metadata = await sharp(img).metadata();
            
            // Upscale by 2.5x to reach roughly 4K quality from 1080p
            const targetWidth = Math.round(metadata.width * 2.5);
            
            await sharp(img)
                .resize(targetWidth, null, {
                    kernel: sharp.kernel.lanczos3,
                    fastShrinkOnLoad: false
                })
                .sharpen(1.5, 0.5, 1.2) // Subtle sharpening pass to mimic EDSR edges
                .jpeg({ quality: 100 })
                .toFile(outPath);
                
            console.log(`Finished ${outPath}`);
        }
    }
}

upscale();
